Modules/xASL_module_Import.m

```diff
@@ -225,3 +228,19 @@
     diary off
 
 end
+
+
+%% Determine the lock dir paths
+function x = xASL_imp_InitLockDirPaths(x,studyPath)
+
+    x.modules.import.dir.lockImport = fullfile(studyPath, 'derivatives', 'ExploreASL', 'lock', 'xASL_module_Import');
```

Comment:
@MichaelStritt This looks different from what `xASL_Iterate` does, I would just follow what `xASL_Iterate` does, and use the same folders this function does (which is by naming these four steps as four submodules, status files, of the module import).

Reply:
@HenkMutsaerts My problem with xASL_Iterate is that AFAIK it basically expects the same subject/session directory structure throughout the pipeline or you have to know it beforehand. That doesn't really work since DCM2NII, NII2BIDS, & BIDS2LEGACY have different subject/session directory structures. If that's not true then let me know.
Right now there is way to much in this single issue though, we should trim it down by a lot. I can try to implement it with xASL_Iterate, but to be honest, I always hated that script 😆

Reply:
Let me have a look if we can make it work with xASL_Iteration if that makes you happy. Most of the time it's super annoying for people during debugging though.